python/taichi/lang/field.py

```diff
@@ -205,7 +205,13 @@ def _pad_key(self, key):
             key = ()
         if not isinstance(key, (tuple, list)):
             key = (key, )
-        assert len(key) == len(self.shape)
+
+        if len(key) != len(self.shape):
+            raise AssertionError(
+                "Slicing an array or matrix from ti.field is not supported so far."
```

Comment:
Shall we simply say Slicing is not supported on ti.field? I'm a bit worried users confuse "matrix" here with the matrix element inside a ti.field ;)